### ISSUE TITLE
[Question]: Can we let the LLM automatically identify whether RAG should be used in different senarios?

### ISSUE THREAD

**Issue by waltonfuture** [run-llama/llama_index#11573]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

For every question, using RAG is time-consuming. I wonder if we can let the LLM automatically identify whether RAG should be used in different senarios.

**Comment by ravi03071991** [comment]
@waltonfuture Yes, you can use a router that routes your query to RAG or the LLM itself can answer it. Esentially the router itself is an LLM.

**Comment by waltonfuture** [comment]
> @waltonfuture Yes, you can use a router that routes your query to RAG or the LLM itself can answer it. Esentially the router itself is an LLM.

Thanks for your advice! But how can the router automatically judge whether my input need RAG or not without human control?

**Comment by durga0201** [comment]
@waltonfuture Routers are modules that take in a user query and a set of “choices” (defined by metadata), and returns one or more selected choices.

They can be used on their own (as “selector modules”), or used as a query engine or retriever (e.g. on top of other query engines/retrievers).

They are simple but powerful modules that use LLMs for decision making capabilities. They can be used for the following use cases and more:

Selecting the right data source among a diverse range of data sources

Deciding whether to do summarization (e.g. using summary index query engine) or semantic search (e.g. using vector index query engine)

Deciding whether to “try” out a bunch of choices at once and combine the results (using multi-routing capabilities).

The core router modules exist in the following forms:

LLM selectors put the choices as a text dump into a prompt and use LLM text completion endpoint to make decisions

Pydantic selectors pass choices as Pydantic schemas into a function calling endpoint, and return Pydantic objects
